thrust/examples/dot_products_with_zip.cu

```diff
@@ -33,10 +31,9 @@ thrust::host_vector<float> random_vector(const size_t N, unsigned int seed = thr
   thrust::default_random_engine rng(seed);
   thrust::uniform_real_distribution<float> u01(0.0f, 1.0f);
   thrust::host_vector<float> temp(N);
-  for (size_t i = 0; i < N; i++)
-  {
-    temp[i] = u01(rng);
-  }
+  thrust::generate(temp.begin(), temp.end(), [&]() {
+    return u01(rng);
+  });
```

Comment:
Critical: this will call `u01::operator()` in parallel which may lead to data races.